### DISCUSSION TITLE
How can I use pagination in 1.4.2?

### DISCUSSION THREAD

**Discussion by mcfl-uri** [vitmalina/w2ui#2186] [Q&A] [answered]
Hi, I currently have a project where I am using 1.4.2, and it is great. However, as I have a ton of records in a remote data source, it would be great if I could use pagination. Is there a way to do so without changing w2ui version? If not, what can I do to improve loading times?

**Comment by mpf82** [comment, marked as answer]
Pagination has been deprecated in w2ui 1.3 in favor of infinite scrolling, see https://w2ui.com/web/blog/8/W2UI-1.3-Released-Goodbye-Pagination-Hello-Infinite-Scroll

So if you really want pagination, you'll have to downgrade to w2ui 1.2 (not recommended)

If you want to upgrade, I've compiled some migration help in this discussion: #2015 

> what can I do to improve loading times

First, identify your bottle-neck. Maybe it's actually your network / backend / DB and not w2ui.

You can also play around with grid.limit to decrease the number of records loaded at once.

**Reply by vitmalina** [reply]
Please note infinite scroll pulls 100 records by default and adds more as you scroll. There is also and option to "skip" some records, or start displaying with a N-th record. It should be available in top ||| button. With this, there should be no need for pagination. 

If pagination is still needed, you could write it yourself and put it in the grid status. It should be pretty straightforward. But I think pagination is an obsolete concept. ))
